### PR TITLE
[SLP]Emit actual bitwidth for analyzed MinBitwidth nodes, NFCI.

### PR DESCRIPTION
**Pull request by alexey-bataev**
SLP includes analysis for the minimum bitwidth, the actual integer
operations can be emitted. It allows to reduce register pressure and
improve perf. Currently, it includes only cost model and the next
transformation relies on InstructionCombiner. Better to do it directly
in SLP, it allows to reduce compile time and fix cost model issues.

### CODE REVIEW

**Review by RKSimon — APPROVED** [review]
LGTM

**Comment by RKSimon** [comment]
FTR: 5adfad254edb765fbbabdf5e7521cfb3111dfeb1

File: llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp [hunk 1]
```diff
@@ -9034,6 +9079,28 @@ InstructionCost BoUpSLP::getTreeCost(ArrayRef<Value *> VectorizedVals) {
             FirstUsers.emplace_back(VU, ScalarTE);
             DemandedElts.push_back(APInt::getZero(FTy->getNumElements()));
             VecId = FirstUsers.size() - 1;
+            auto It = MinBWs.find(EU.Scalar);
+            if (It != MinBWs.end() && VectorCasts.insert(EU.Scalar).second) {
+              unsigned BWSz = It->second.second;
```
Comment (valerydmit) [inline llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:9084]: It->second.first ? "second" is  a boolean value.

Comment (alexey-bataev) [reply llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:9084]: Yep, will fix it.

**Pull request closed without merging**